### PR TITLE
Support en passant capture logic for pawns

### PR DESCRIPTION
**Pull request by justinrjoseph**
- Added allows_en_passant_capture? attribute to Game model
- Added en_passant_capture_opportunity virtual attribute and coinciding en_passant_opportunity_active? method to Game model
- Added en_passant_move?(x, y) and coinciding capture logic method to Pawn model (requires capture logic update PR)
- Added logic to pieces controller to handle whether en passant capture opportunity exists or not
- Added integration tests for en passant capture logic

**NOTE: this depends on the update capture logic PR**

### CODE REVIEW

**Comment by justinrjoseph** [comment]
I'll update this PR to resolve the conflict ASAP.

**Comment by justinrjoseph** [comment]
As of 03/30/2016
- Resolved pieces controller conflict
- Added king pieces to en_passant_capture_test setup to satisfy check logic

**TO DO**:
- Double check implementation before requesting review

**Comment by justinrjoseph** [comment]
As of 04/02/2016

pawn.rb
- Cleaned up some indentation
- Added necessary return true statements to en_passant_move? method

**Comment by justinrjoseph** [comment]
I will resolve the current conflict ASAP.

File: app/controllers/pieces_controller.rb [hunk 1]
```diff
@@ -12,10 +12,16 @@ def update
       flash[:danger] = "#{@piece.color.capitalize} #{@piece.type} move to X#{x}/Y#{y} was invalid"
     end
 
+    if @piece.game.en_passant_opportunity_active?
+      @piece.game.update_attribute(:allows_en_passant_capture?, true)
+    else
+      @piece.game.update_attribute(:allows_en_passant_capture?, false) if @piece.game.allows_en_passant_capture?
+    end
+
```
Comment (xJIsaac) [inline app/controllers/pieces_controller.rb]: Just curious why this logic above was added to a controller and not a model.

Comment (justinrjoseph) [inline app/controllers/pieces_controller.rb]: Your question is a bit vague. What exactly are you suggesting? The only comment I can offer right now is all the code in pieces#update is there to represent the various things that can/should happen within the context of a player turn.

I'll explain/elaborate further once I have a better understanding of what you're asking about.

Comment (xJIsaac) [inline app/controllers/pieces_controller.rb]: Generally, I've seen conditional logic for updating the db in models. I've understood it was best practice to use the controller to authenticate the request but the model is ultimately the device used to make the change if needed. This was something that was pointed out to me by my mentor although I can't remember the reason why this was considered best practice. So like I said I was just curious about it. I have no critique for your code. Just want to better understand it as it stood out from how I normally see controllers written.

Comment (justinrjoseph) [inline app/controllers/pieces_controller.rb]: Alright, finally some time to respond :)

Yeah, I wouldn't be surprised if there was a better, 'more appropriate' way to handle this logic as far as where to put it and what have you. I don't know if there is, but I'm happy to look at it. Might have an opportunity this weekend, we'll see.

If it makes sense to move it into the Game model, I'll give it a go. Don't know what would or wouldn't work at the moment.

Comment (justinrjoseph) [inline app/controllers/pieces_controller.rb]: Well, I guess you made me curious enough. I moved the logic in question to the piece move_to! method before the final return of true, and it seems to work fine. Not bad.

**Pull request merged**